cmd/serverless-init/tag/tag.go

```diff
@@ -68,3 +68,23 @@ func GetBaseTagsArrayWithMetadataTags(metadata map[string]string) []string {
 	}
 	return tagsArray
 }
+
+func ArrayTagToMap(tagArray []string) map[string]string {
+	toMap := make(map[string]string)
+	for _, v := range tagArray {
+		keyVal := strings.Split(v, ":")
+		toMap[keyVal[0]] = keyVal[1]
+	}
+	return toMap
+}
+
+func MergeWithOverwrite(tags map[string]string, overwritingTags map[string]string) map[string]string {
+	merged := make(map[string]string)
+	for k, v := range tags {
+		merged[k] = v
+	}
+	for k, v := range overwritingTags {
```

Comment:
I may be mistaken, but I think we don't overwrite user tags anywhere else in the agent, and if the same tag key occurs multiple times with different values, we will report all instances of the tag. Could you please provide a bit of context why this is necessary in serverless?

Reply:
Oh right, thanks for catching it.
There is also some corner cases like if a value is defined as boolean true/false, where it might not follow the append-value rule. I will try to find the original code and reuse that

Reply:
I tried to run the tests of the tracer as a guideline
https://github.com/DataDog/dd-trace-js/blob/39656348312aee8eb62be8fa260212772bf6e997/packages/dd-trace/test/config.spec.js#L469

and they support multiple-column values indeed.  will fix it.
For the duplicate keys, it seems that the value is overwritten and not concatenated.
This case is handling the `DD_TAGS` and `DD_EXTRA_TAGS` only, so the cardinality might be of only one value per key.

Does it make sense or should it be investigated further?

Reply:
> For the duplicate keys, it seems that the value is overwritten

Could you please provide a link to where this happens in the agent?

Reply:
https://github.com/DataDog/datadog-agent/blob/main/pkg/serverless/tags/tags.go#L94-L99